pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -138,6 +146,12 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, err
 	}
 
+	meta.SetStatusCondition(&{{ lower .Resource.Kind }}.Status.Conditions, metav1.Condition{Type: typeProgressing{{ .Resource.Kind }}, Status: metav1.ConditionTrue, Reason: "Reconciling", Message: "Starting reconciliation"})
```

Comment:
@NikhilSharmaWe : 

```suggestion
	meta.SetStatusCondition(&{{ lower .Resource.Kind }}.Status.Conditions, metav1.Condition{Type: typeAvailable{{ .Resource.Kind }},, Reason: "Reconciling", Message: "Starting reconciliation"})
```

See from the API convention : 

> Controllers should apply their conditions to a resource the first time they visit the resource, even if the status is Unknown. This allows other components in the system to know that the condition exists and the controller is making progress on reconciling that resource.

> Not all controllers will observe the previous advice about reporting "Unknown" or "False" values. For known conditions, the absence of a condition status should be interpreted the same as Unknown, and typically indicates that reconciliation has not yet finished (or that the resource state may not yet be observable).

Also, before to add this status we need first check to have we status condition Available already? If not then we add.

Reply:
@NikhilSharmaWe the setStatus implementation already check if that exist or not so we just need to use the set directly without the need to check, see: https://github.com/kubernetes/apimachinery/blob/v0.24.3/pkg/api/meta/conditions.go#L30